cmd/cagent/main.go

```diff
@@ -330,12 +330,22 @@ func main() {
 	interruptChan := make(chan struct{})
 	doneChan := make(chan struct{})
 
+	if ca.PidFile != "" && *oneRunOnlyModePtr && runtime.GOOS != "windows" {
+		err := ioutil.WriteFile(ca.PidFile, []byte(strconv.Itoa(os.Getpid())), 0664)
+		if err != nil {
+			log.Errorf("Failed to write pid file at: %s", ca.PidFile)
+		}
+	}
+
 	if *oneRunOnlyModePtr == true {
-		ca.Run(output, interruptChan, true)
+		err := ca.RunOnce(output)
+		if err != nil {
+			log.Fatal(err)
+		}
 		return
 	} else {
```

Comment:
the `else` can be removed.
Inside the `if` branch either `log.Fatal` or `return` is called which end execution. Therefor the `else` can be omitted

Reply:
thx, fixed